src/pages/ReimbursementAccount/ReimbursementAccountPage.js

```diff
@@ -97,447 +95,397 @@ const defaultProps = {
+                    BankAccounts.goToWithdrawalAccountSetupStep(CONST.BANK_ACCOUNT.STEP.ACH_CONTRACT);
+                } else if (!isOffline && achData.state === BankAccount.STATE.PENDING) {
+                    setShouldShowContinueSetupButton(true);
+                } else {
+                    Navigation.goBack(backTo);
+                }
+                break;
+
             default:
-                return [];
+                Navigation.goBack(backTo);
         }
-    }
+    };
 
     /**
-     * @param {Array} fieldNames
-     *
-     * @returns {*}
+     * Returns true if a VBBA exists in any state other than OPEN or LOCKED
+     * @returns {Boolean}
      */
-    getBankAccountFields(fieldNames) {
-        return {
-            ..._.pick(lodashGet(this.props.reimbursementAccount, 'achData'), ...fieldNames),
-        };
-    }
+    function hasInProgressVBBA() {
+        const achData = lodashGet(reimbursementAccount, 'achData', {});
+        return achData.bankAccountID
+            && achData.state !== BankAccount.STATE.OPEN
+            && achData.state !== BankAccount.STATE.LOCKED;
+    };
 
     /*
      * Calculates the state used to show the "Continue with setup" view. If a bank account setup is already in progress and
      * no specific further step was passed in the url we'll show the workspace bank account reset modal if the user wishes to start over
      */
-    getShouldShowContinueSetupButtonInitialValue() {
-        if (!this.hasInProgressVBBA()) {
+    function getShouldShowContinueSetupButtonInitialValue() {
+        if (!hasInProgressVBBA()) {
             // Since there is no VBBA in progress, we won't need to show the component ContinueBankAccountSetup
             return false;
         }
-        const achData = lodashGet(this.props.reimbursementAccount, 'achData', {});
-        return achData.state === BankAccount.STATE.PENDING || _.contains([CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT, ''], this.getStepToOpenFromRouteParams());
-    }
-
-    /**
-     * @param {String} fieldName
-     * @param {*} defaultValue
-     *
-     * @returns {*}
-     */
-    getDefaultStateForField(fieldName, defaultValue = '') {
-        return lodashGet(this.props.reimbursementAccount, ['achData', fieldName], defaultValue);
+        const achData = lodashGet(reimbursementAccount, 'achData', {});
+        return achData.state === BankAccount.STATE.PENDING || _.contains([CONST.BANK_ACCOUNT.STEP.BANK_ACCOUNT, ''], getStepToOpenFromRouteParams());
     }
 
     /**
      * We can pass stepToOpen in the URL to force which step to show.
      * Mainly needed when user finished the flow in verifying state, and Ops ask them to modify some fields from a specific step.
      * @returns {String}
      */
-    getStepToOpenFromRouteParams() {
-        switch (lodashGet(this.props.route, ['params', 'stepToOpen'], '')) {
-            case 'new':
+    function getStepToOpenFromRouteParams() {
```

Comment:
Looks like this function can be moved out of component, just pass one param. If it is possible, try to reduce body  of the component to avoid re-creating functions for each re-render